strawberry/sanic/views.py

```diff
@@ -62,29 +64,48 @@ async def dispatch_request(self, request: Request):  # type: ignore
             template = render_graphiql_page()
             return self.render_template(template=template)
 
-        data = request.json
-        try:
-            query = data["query"]
-        except KeyError:
-            return HTTPResponse("No valid query was provided for the request", 400)
-
-        variables = data.get("variables")
-        operation_name = data.get("operationName")
+        operation_context = self.get_execution_context(request)
         context = await self.get_context(request)
+        root_value = self.get_root_value()
 
         result = await self.schema.execute(
-            query,
-            variable_values=variables,
+            query=operation_context.query,
+            variable_values=operation_context.variables,
             context_value=context,
-            operation_name=operation_name,
-            root_value=self.get_root_value(),
+            root_value=root_value,
+            operation_name=operation_context.operation_name,
         )
         response_data = self.process_result(result)
 
         return HTTPResponse(
             json.dumps(response_data), status=200, content_type="application/json"
         )
 
+    def get_execution_context(self, request: Request) -> ExecutionContext:
+        try:
+            data = self.parse_body(request)
+        except json.JSONDecodeError:
+            raise ServerError("Unable to parse request body as JSON", status_code=400)
+
+        try:
+            query = data["query"]
+            variables = data.get("variables")
+            operation_name = data.get("operationName")
```

Comment:
Nitpick: can these be moved out of the `try/except` as they cannot raise a `KeyError`?